### PR TITLE
Cucumber step definitions

### PR DESCRIPTION
**Pull request by josepajay**
Review and merge after #12 

Definitions for cucumber steps except for the final validation part

### CODE REVIEW

File: csv-validation-tests/src/test/scala/steps/StepDefinitions.scala [hunk 1]
```diff
@@ -1,61 +1,88 @@
...
+        case r if r.uri.path.endsWith(notFountEndsWith) =>
+          Response("Not found", StatusCode.NotFound)
+        case r if r.uri.path.endsWith(notFoundStartsWith) =>
+        Response("Not found", StatusCode.NotFound)
+      })
+  }
 
   Given("""^I have a CSV file called "(.*?)"$""") { (filename: String) =>
-    //    log.info("Log me here..") Not working, need to fix
     val filePath = fixturesPath + filename
```
Comment (robons) [inline csv-validation-tests/src/test/scala/steps/StepDefinitions.scala:43]: Shouldn't these be the class level properties you're setting?

File: csv-validation-tests/src/test/scala/steps/StepDefinitions.scala [hunk 2]
```diff
@@ -1,61 +1,88 @@
...
-//  And("there should not be warnings") { () =>
-//    // define
-//  }
+    // notFoundStartsWith is a list which holds all the url startwith strings for which a 404 should be returned.
+    // This list is later used in setTestingBackend function to mock http calls
+    notFoundStartsWith = url :: notFoundStartsWith
+  }
+
+  Given("""^I have a metadata file called "([^"]*)"$""") { filename: String =>
+    val filePath = fixturesPath + filename
```
Comment (robons) [inline csv-validation-tests/src/test/scala/steps/StepDefinitions.scala:63]: Again, shouldn't you be setting class properties here?

File: csv-validation-tests/src/test/scala/steps/StepDefinitions.scala [hunk 3]
```diff
@@ -1,61 +1,88 @@
...
+    val metadata = Source.fromFile(filePath).getLines.mkString
+  }
+
+  And("""^the (schema|metadata) is stored at the url "(.*?)"$""") { (schemaType:String, url:String) =>
+    schemaUrl = url
+  }
 
+  And("""^I have a file called "(.*?)" at the url "(.*?)"$""") { (fileName: String, url:String) =>
+    val filePath = fixturesPath + fileName
```
Comment (robons) [inline csv-validation-tests/src/test/scala/steps/StepDefinitions.scala:72]: Class level properties?

Comment (josepajay) [reply csv-validation-tests/src/test/scala/steps/StepDefinitions.scala:72]: filePath is only applicable for the specific step definition and I don't think these must be made class level property. Whereas I have defined some other properties (like csv, metadata inside the step definition) which I have changed to class properties. 
Example scenario steps
		Given I have a CSV file called "csvw/test001.csv"
		And it is stored at the url "https://w3c.github.io/csvw/tests/test001.csv"
		And there is no file at the url "https://w3c.github.io/.well-known/csvm"

**Review by robons — APPROVED** [review]

**Pull request merged**